api/registry.go

```diff
@@ -6,143 +6,88 @@ import (
+	})
+	router.GET("/v1/" + handle + "/xpub/:xpub", func(c *gin.Context) {
+		endpoint.GetTransactionsByXpub(c, api)
+	})
+}
 
-	root.GET("/v2/"+handle+"/staking/validators", middleware.CacheMiddleware(time.Hour, func(c *gin.Context) {
-		endpoint.GetValidators(c, stakingAPI)
+func RegisterStakeAPI(router gin.IRouter, api blockatlas.StakeAPI) {
+	handle := api.Coin().Handle
+	router.GET("/v2/" + handle + "/staking/validators", middleware.CacheMiddleware(time.Hour, func(c *gin.Context) {
+		endpoint.GetValidators(c, api)
 	}))
-	root.GET("/v2/"+handle+"/staking/delegations/:address", func(c *gin.Context) {
-		endpoint.GetStakingDelegationsForSpecificCoin(c, stakingAPI)
+	router.GET("/v2/" + handle + "/staking/delegations/:address", func(c *gin.Context) {
+		endpoint.GetStakingDelegationsForSpecificCoin(c, api)
 	})
 }
 
-func RegisterBatchAPI(root gin.IRouter) {
-	root.POST("v2/staking/delegations", func(c *gin.Context) {
+func RegisterBatchAPI(router gin.IRouter) {
+	router.POST("v2/staking/delegations", func(c *gin.Context) {
 		endpoint.GetStakeDelegationsWithAllInfoForBatch(c, platform.StakeAPIs)
 	})
-	root.POST("v2/staking/list", middleware.CacheMiddleware(time.Hour, func(c *gin.Context) {
+	router.POST("v2/staking/list", middleware.CacheMiddleware(time.Hour, func(c *gin.Context) {
 		endpoint.GetStakeInfoForBatch(c, platform.StakeAPIs)
 	}))
-	root.POST("/v3/collectibles/categories", func(c *gin.Context) {
-		endpoint.GetCollectionCategoriesFromListV3(c, platform.CollectionAPIs)
+	router.POST("/v3/collectibles/categories", func(c *gin.Context) {
+		endpoint.GetCollectionCategoriesFromListV3(c, platform.CollectionsAPIs)
 	})
-	root.POST("/v4/collectibles/categories", func(c *gin.Context) {
-		endpoint.GetCollectionCategoriesFromList(c, platform.CollectionAPIs)
+	router.POST("/v4/collectibles/categories", func(c *gin.Context) {
+		endpoint.GetCollectionCategoriesFromList(c, platform.CollectionsAPIs)
 	})
-	root.POST("/v2/tokens", func(c *gin.Context) {
+	router.POST("/v2/tokens", func(c *gin.Context) {
 		endpoint.GetTokens(c, platform.TokensAPIs)
 	})
 }
 
-// CustomAPI must be removed and all handlers needs to be migrated to the transactions, tokens api
-func RegisterCustomAPI(root gin.IRouter, api blockatlas.Platform) {
-	customAPI, ok := api.(blockatlas.CustomAPI)
-	if !ok {
-		return
-	}
-	handle := api.Coin().Handle
-
-	customRouter := root.Group("/v1/" + handle)
-	customAPI.RegisterRoutes(customRouter)
-}
-
-func RegisterDomainAPI(root gin.IRouter) {
-	root.GET("/ns/lookup", endpoint.GetAddressByCoinAndDomain)
-	root.GET("v2/ns/lookup", endpoint.GetAddressByCoinAndDomainBatch)
-}
-
-func RegisterBasicAPI(root gin.IRouter) {
-	root.GET("/", endpoint.GetStatus)
-	root.GET("/metrics", ginprom.PromHandler(promhttp.Handler()))
+func RegisterDomainAPI(router gin.IRouter) {
+	router.GET("/ns/lookup", endpoint.GetAddressByCoinAndDomain)
+	router.GET("v2/ns/lookup", endpoint.GetAddressByCoinAndDomainBatch)
```

Comment:
`/v2/ns/lookup/`

Reply:
I added starting / to 3 routes that missed it, but I'm not sure about the trailing slashes, that's missing in many routes.  What's the significance of trailing slash?